app/src/main/java/com/fluffybros/tujefe/MainActivity.kt

```diff
@@ -25,4 +33,59 @@ class MainActivity : AppCompatActivity() {
         setupActionBarWithNavController(navController, appBarConfiguration)
         navView.setupWithNavController(navController)
     }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (resultCode == Activity.RESULT_OK) {
+            val imageBitmap = data?.extras?.get("data") as Bitmap
+            val barcode = decodeBitmap(imageBitmap)
+            if (barcode == null) {
+                // TODO: let user know nothing was detected
+                return
+            }
+            // Parse the data
+            if (barcode.displayValue.contains("secret=")) {
+                val secretSearch = ArrayList<String>()
```

Comment:
you can just inline this - `.findAnyOf(arrayListOf("secret="), 0, false)`
do this everywhere else as well